pulseapi/profiles/models.py

```diff
@@ -23,9 +121,72 @@ class UserProfile(models.Model):
         through='profiles.UserBookmarks'
     )
 
+    # The name field is an alternative name to be used if the
+    # associated user(s) don't want to expose their login-indicated
+    # name instead.
+    #
+    # Examples of this are nicknames, pseudonyms, and org names
+    custom_name = models.CharField(
+        max_length=500,
+        blank=True,
+        null=True
+    )
+    custom_name.short_description = 'Custom user name'
+
+    # Accessing the Profile-indicated name needs various stages
+    # of fallback.
+    def name(self):
+        if not self.custom_name:
```

Comment:
the problem with using `not` here is that `if not "   "` evaluates to `False` so it will return the whitespaces. Don't know if that is expected behavior or not.